<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Generates heatbox and scatter plots from a feature matrix. • corplot</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Generates heatbox and scatter plots from a feature matrix.">
<meta property="og:description" content="Genomics data often stored in a matrix like format, where each row is a feature (gene, transcript, protein etc.) and column is a variable (e.g. expression value in RNA-seq, ChIP-seq, Pol-II ChIP-seq etc.). Variables are often grouped by replicates, time-point or specific experimental conditions such as wild type, delition, control, treatment etc. In such a multidimental data, plotting a x-y scatter plot between diffrent groups require lots of data wrangling before we input them for final ggplot.
    corplot has functions to generate heatbox and pairwise scatterplots directly from feature matix given in a tbl format. Let’s have a look into required input data and resultant plots outof corpot.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">corplot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cparsania/corplot/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="corplot" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#corplot" class="anchor"></a>corplot</h1></div>
<!-- badges: start -->

<div id="motivation" class="section level2">
<h2 class="hasAnchor">
<a href="#motivation" class="anchor"></a>Motivation</h2>
<p>Genomics data often stored in a matrix like format, where each row is a feature (gene, transcript, protein etc.) and columns are variables (e.g. signal intensity of experiments such as RNA-seq, ChIP-seq, Pol-II ChIP-seq etc.). Variables are often grouped by replicates, time-points or specific experimental conditions such as wild type, deletion, control, treatment etc. In such a multidimensional data, plotting a x-y scatter plot between different groups require lots of data wrangling before it goes for final ggplot.</p>
<p><code>corplot</code> has functions to generate heatbox and pairwise scatter plots directly from feature matrix given in a tbl format. Let’s have a look into required input data and resultant plots out of <code>corpot</code>.</p>
</div>
<div id="install" class="section level2">
<h2 class="hasAnchor">
<a href="#install" class="anchor"></a>Install</h2>
<div class="sourceCode" id="cb1"><pre class="r"><span class="kw">if</span>(<span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="st">"devtools"</span>)){
        <span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"cparsania/corplot"</span>)
} <span class="kw">else</span>{
        <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"devtools"</span>)
        <span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"cparsania/corplot"</span>)
}</pre></div>
</div>
<div id="correlation-heatbox" class="section level2">
<h2 class="hasAnchor">
<a href="#correlation-heatbox" class="anchor"></a>Correlation heatbox</h2>
</div>
<div id="all-samples-vs-all-samples" class="section level2">
<h2 class="hasAnchor">
<a href="#all-samples-vs-all-samples" class="anchor"></a>All samples vs all samples</h2>
<div class="sourceCode" id="cb2"><pre class="r">
<span class="no">expr_mat_file</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span> ,<span class="st">"example_data_expr_mat_01.txt"</span> , <span class="kw">package</span> <span class="kw">=</span> <span class="st">"corplot"</span>)
<span class="no">expr_mat</span> <span class="kw">&lt;-</span> <span class="kw pkg">readr</span><span class="kw ns">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_delim</a></span>(<span class="no">expr_mat_file</span> , <span class="kw">delim</span> <span class="kw">=</span> <span class="st">"\t"</span>)

<span class="no">expr_mat</span>
<span class="co">#&gt; # A tibble: 6,338 x 9</span>
<span class="co">#&gt;    gene_name Control_Rep.A Control_Rep.B Treat1_Rep.A Treat1_Rep.B Treat2_Rep.A</span>
<span class="co">#&gt;    &lt;chr&gt;             &lt;dbl&gt;         &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;</span>
<span class="co">#&gt;  1 C1_00010…          1             0            1            2.81         5.13</span>
<span class="co">#&gt;  2 C1_00020…          9.65          9.32         9.15         9.32        10.9 </span>
<span class="co">#&gt;  3 C1_00030…          5.46          4.70         4.64         5.36         6.15</span>
<span class="co">#&gt;  4 C1_00040…         10.9          10.5         11.4         12.2         11.3 </span>
<span class="co">#&gt;  5 C1_00050…          0             1            1            5.88         5.43</span>
<span class="co">#&gt;  6 C1_00060…         13.4          13.0         12.9         13.1         13.2 </span>
<span class="co">#&gt;  7 C1_00070…         12.9          12.7         12.6         12.2         11.2 </span>
<span class="co">#&gt;  8 C1_00080…         10.4           9.81        10.2          9.96        10.4 </span>
<span class="co">#&gt;  9 C1_00090…          7.33          6.61         6.13         6.88         7   </span>
<span class="co">#&gt; 10 C1_00100…         10.0          10.1         10.9         10.9         10.8 </span>
<span class="co">#&gt; # … with 6,328 more rows, and 3 more variables: Treat2_Rep.B &lt;dbl&gt;,</span>
<span class="co">#&gt; #   Treat3_Rep.A &lt;dbl&gt;, Treat3_Rep.B &lt;dbl&gt;</span>

<span class="co">## calculate  pairwise correlation </span>
<span class="no">cor_tbl</span> <span class="kw">&lt;-</span> <span class="kw pkg">corplot</span><span class="kw ns">::</span><span class="fu"><a href="reference/get_pairwise_cor_tbl.html">get_pairwise_cor_tbl</a></span>(<span class="no">expr_mat</span> , <span class="kw">var</span> <span class="kw">=</span> <span class="st">"gene_name"</span> , <span class="kw">method</span> <span class="kw">=</span> <span class="st">"pearson"</span>)

<span class="no">cor_tbl</span>
<span class="co">#&gt; # A tibble: 64 x 3</span>
<span class="co">#&gt;    var1          var2           corr</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;chr&gt;         &lt;dbl&gt;</span>
<span class="co">#&gt;  1 Control_Rep.A Control_Rep.A  1   </span>
<span class="co">#&gt;  2 Control_Rep.B Control_Rep.A  0.96</span>
<span class="co">#&gt;  3 Treat1_Rep.A  Control_Rep.A  0.93</span>
<span class="co">#&gt;  4 Treat1_Rep.B  Control_Rep.A  0.9 </span>
<span class="co">#&gt;  5 Treat2_Rep.A  Control_Rep.A  0.86</span>
<span class="co">#&gt;  6 Treat2_Rep.B  Control_Rep.A  0.86</span>
<span class="co">#&gt;  7 Treat3_Rep.A  Control_Rep.A  0.91</span>
<span class="co">#&gt;  8 Treat3_Rep.B  Control_Rep.A  0.91</span>
<span class="co">#&gt;  9 Control_Rep.A Control_Rep.B  0.96</span>
<span class="co">#&gt; 10 Control_Rep.B Control_Rep.B  1   </span>
<span class="co">#&gt; # … with 54 more rows</span>

<span class="no">cp</span> <span class="kw">&lt;-</span> <span class="kw pkg">corplot</span><span class="kw ns">::</span><span class="fu"><a href="reference/get_corr_heat_box.html">get_corr_heat_box</a></span>(<span class="no">cor_tbl</span>,<span class="kw">var1</span> <span class="kw">=</span> <span class="no">var1</span>, <span class="kw">var2</span> <span class="kw">=</span> <span class="no">var2</span> ,<span class="kw">value</span> <span class="kw">=</span> <span class="no">corr</span>)
<span class="no">cp</span> + <span class="kw pkg">viridis</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/viridis/man/scale_viridis.html">scale_fill_viridis</a></span>() + <span class="kw pkg">ggplot2</span><span class="kw ns">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="kw">axis.text.x</span> <span class="kw">=</span> <span class="kw pkg">ggplot2</span><span class="kw ns">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="kw">angle</span><span class="kw">=</span><span class="fl">90</span>))</pre></div>
<p><img src="reference/figures/README-unnamed-chunk-5-1.png" width="100%"></p>
</div>
<div id="group-by-replicates" class="section level2">
<h2 class="hasAnchor">
<a href="#group-by-replicates" class="anchor"></a>Group by replicates</h2>
<p>All samples vs all samples correlation heatbox has redundant samples on each axis. This makes plot less readable. Alternate way to overcome this is to plot samples of replicate 1 vs samples of replicate 2.</p>
<div class="sourceCode" id="cb3"><pre class="r"><span class="no">cor_tbl2</span> <span class="kw">&lt;-</span> <span class="no">cor_tbl</span> <span class="kw">%&gt;%</span> <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">"Rep.A"</span>, <span class="no">var1</span>) ) <span class="kw">%&gt;%</span>  <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">"Rep.B"</span>, <span class="no">var2</span>) )

<span class="no">cor_tbl2</span>
<span class="co">#&gt; # A tibble: 16 x 3</span>
<span class="co">#&gt;    var1          var2           corr</span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;chr&gt;         &lt;dbl&gt;</span>
<span class="co">#&gt;  1 Control_Rep.A Control_Rep.B  0.96</span>
<span class="co">#&gt;  2 Treat1_Rep.A  Control_Rep.B  0.95</span>
<span class="co">#&gt;  3 Treat2_Rep.A  Control_Rep.B  0.86</span>
<span class="co">#&gt;  4 Treat3_Rep.A  Control_Rep.B  0.91</span>
<span class="co">#&gt;  5 Control_Rep.A Treat1_Rep.B   0.9 </span>
<span class="co">#&gt;  6 Treat1_Rep.A  Treat1_Rep.B   0.94</span>
<span class="co">#&gt;  7 Treat2_Rep.A  Treat1_Rep.B   0.93</span>
<span class="co">#&gt;  8 Treat3_Rep.A  Treat1_Rep.B   0.9 </span>
<span class="co">#&gt;  9 Control_Rep.A Treat2_Rep.B   0.86</span>
<span class="co">#&gt; 10 Treat1_Rep.A  Treat2_Rep.B   0.9 </span>
<span class="co">#&gt; 11 Treat2_Rep.A  Treat2_Rep.B   0.99</span>
<span class="co">#&gt; 12 Treat3_Rep.A  Treat2_Rep.B   0.9 </span>
<span class="co">#&gt; 13 Control_Rep.A Treat3_Rep.B   0.91</span>
<span class="co">#&gt; 14 Treat1_Rep.A  Treat3_Rep.B   0.92</span>
<span class="co">#&gt; 15 Treat2_Rep.A  Treat3_Rep.B   0.9 </span>
<span class="co">#&gt; 16 Treat3_Rep.A  Treat3_Rep.B   0.97</span>

<span class="kw pkg">corplot</span><span class="kw ns">::</span><span class="fu"><a href="reference/get_corr_heat_box.html">get_corr_heat_box</a></span>(<span class="no">cor_tbl2</span>,<span class="kw">var1</span> <span class="kw">=</span> <span class="no">var1</span>, <span class="kw">var2</span> <span class="kw">=</span> <span class="no">var2</span>, <span class="kw">value</span> <span class="kw">=</span> <span class="no">corr</span>) +
  <span class="kw pkg">viridis</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/viridis/man/scale_viridis.html">scale_fill_viridis</a></span>()</pre></div>
<p><img src="reference/figures/README-unnamed-chunk-6-1.png" width="100%"></p>
</div>
<div id="scatter-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#scatter-plot" class="anchor"></a>Scatter plot</h2>
</div>
<div id="group-by-replicates--all-combinations" class="section level2">
<h2 class="hasAnchor">
<a href="#group-by-replicates--all-combinations" class="anchor"></a>Group by replicates : All combinations</h2>
<div class="sourceCode" id="cb4"><pre class="r"><span class="no">groups_file</span> <span class="kw">&lt;-</span> <span class="no">expr_mat_file</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span> ,<span class="st">"example_data_01_sample_groups.txt"</span> , <span class="kw">package</span> <span class="kw">=</span> <span class="st">"corplot"</span>)
<span class="no">groups</span> <span class="kw">&lt;-</span> <span class="kw pkg">readr</span><span class="kw ns">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_delim</a></span>(<span class="kw">file</span> <span class="kw">=</span> <span class="no">groups_file</span>,<span class="kw">delim</span> <span class="kw">=</span> <span class="st">"\t"</span>)

<span class="no">groups</span>
<span class="co">#&gt; # A tibble: 8 x 3</span>
<span class="co">#&gt;   samples       condition repl </span>
<span class="co">#&gt;   &lt;chr&gt;         &lt;chr&gt;     &lt;chr&gt;</span>
<span class="co">#&gt; 1 Control_Rep.A Control   Rep.A</span>
<span class="co">#&gt; 2 Control_Rep.B Control   Rep.B</span>
<span class="co">#&gt; 3 Treat1_Rep.A  Treat1    Rep.A</span>
<span class="co">#&gt; 4 Treat1_Rep.B  Treat1    Rep.B</span>
<span class="co">#&gt; 5 Treat2_Rep.A  Treat2    Rep.A</span>
<span class="co">#&gt; 6 Treat2_Rep.B  Treat2    Rep.B</span>
<span class="co">#&gt; 7 Treat3_Rep.A  Treat3    Rep.A</span>
<span class="co">#&gt; 8 Treat3_Rep.B  Treat3    Rep.B</span>

<span class="no">csp</span> <span class="kw">&lt;-</span> <span class="kw pkg">corplot</span><span class="kw ns">::</span><span class="fu"><a href="reference/get_pair_wise_scatter.html">get_pair_wise_scatter</a></span>(<span class="kw">dat_tbl</span> <span class="kw">=</span> <span class="no">expr_mat</span>, <span class="kw">group_tbl</span> <span class="kw">=</span> <span class="no">groups</span>,<span class="kw">var_plot</span> <span class="kw">=</span> <span class="no">condition</span>, <span class="kw">var_plot_group</span> <span class="kw">=</span> <span class="no">repl</span>,<span class="kw">dat_id</span> <span class="kw">=</span> <span class="no">gene_name</span>)

<span class="no">csp</span></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-7-1.png" width="100%"></p>
<div id="display-corr-value" class="section level3">
<h3 class="hasAnchor">
<a href="#display-corr-value" class="anchor"></a>Display corr value</h3>
<div class="sourceCode" id="cb5"><pre class="r">
<span class="no">cor_tbl2</span> <span class="kw">&lt;-</span> <span class="no">cor_tbl</span> <span class="kw">%&gt;%</span> <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">rename</a></span>(<span class="kw">`Rep.A`</span><span class="kw">=</span><span class="no">var1</span>, <span class="kw">`Rep.B`</span> <span class="kw">=</span> <span class="no">var2</span>) <span class="kw">%&gt;%</span>
  <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">"Rep.A"</span> ,<span class="no">`Rep.A`</span>)) <span class="kw">%&gt;%</span>
  <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span>(<span class="st">"Rep.B"</span> ,<span class="no">`Rep.B`</span>)) <span class="kw">%&gt;%</span>
  <span class="kw pkg">TidyWrappers</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/TidyWrappers/man/tbl_replace_string.html">tbl_replace_string</a></span>(<span class="st">"_.*"</span> , <span class="st">""</span>)

<span class="no">cor_tbl2</span>
<span class="co">#&gt; # A tibble: 16 x 3</span>
<span class="co">#&gt;    Rep.A   Rep.B    corr</span>
<span class="co">#&gt;    &lt;chr&gt;   &lt;chr&gt;   &lt;dbl&gt;</span>
<span class="co">#&gt;  1 Control Control  0.96</span>
<span class="co">#&gt;  2 Treat1  Control  0.95</span>
<span class="co">#&gt;  3 Treat2  Control  0.86</span>
<span class="co">#&gt;  4 Treat3  Control  0.91</span>
<span class="co">#&gt;  5 Control Treat1   0.9 </span>
<span class="co">#&gt;  6 Treat1  Treat1   0.94</span>
<span class="co">#&gt;  7 Treat2  Treat1   0.93</span>
<span class="co">#&gt;  8 Treat3  Treat1   0.9 </span>
<span class="co">#&gt;  9 Control Treat2   0.86</span>
<span class="co">#&gt; 10 Treat1  Treat2   0.9 </span>
<span class="co">#&gt; 11 Treat2  Treat2   0.99</span>
<span class="co">#&gt; 12 Treat3  Treat2   0.9 </span>
<span class="co">#&gt; 13 Control Treat3   0.91</span>
<span class="co">#&gt; 14 Treat1  Treat3   0.92</span>
<span class="co">#&gt; 15 Treat2  Treat3   0.9 </span>
<span class="co">#&gt; 16 Treat3  Treat3   0.97</span>

<span class="no">csp</span> + <span class="kw pkg">ggplot2</span><span class="kw ns">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span>(<span class="kw">data</span> <span class="kw">=</span> <span class="no">cor_tbl2</span>,  <span class="kw">x</span> <span class="kw">=</span> <span class="fl">4</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="fl">18</span>, <span class="kw pkg">ggplot2</span><span class="kw ns">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">label</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"r"</span>,<span class="st">"="</span>,<span class="no">corr</span> , <span class="kw">sep</span> <span class="kw">=</span> <span class="st">""</span>)) ,
                         <span class="kw">fontface</span><span class="kw">=</span><span class="st">"italic"</span> , <span class="kw">col</span> <span class="kw">=</span> <span class="st">"red"</span>,<span class="kw">size</span> <span class="kw">=</span> <span class="fl">5</span>)</pre></div>
<p><img src="reference/figures/README-unnamed-chunk-8-1.png" width="100%"></p>
</div>
</div>
<div id="group-by-replicates--only-replicate-pairs" class="section level2">
<h2 class="hasAnchor">
<a href="#group-by-replicates--only-replicate-pairs" class="anchor"></a>Group by replicates : Only replicate pairs</h2>
<div class="sourceCode" id="cb6"><pre class="r"><span class="no">csp2</span> <span class="kw">&lt;-</span> <span class="kw pkg">corplot</span><span class="kw ns">::</span><span class="fu"><a href="reference/get_pair_wise_scatter.html">get_pair_wise_scatter</a></span>(<span class="kw">dat_tbl</span> <span class="kw">=</span> <span class="no">expr_mat</span>, <span class="kw">group_tbl</span> <span class="kw">=</span> <span class="no">groups</span>,<span class="kw">var_plot</span> <span class="kw">=</span> <span class="no">condition</span>, <span class="kw">var_plot_group</span> <span class="kw">=</span> <span class="no">repl</span>,<span class="kw">dat_id</span> <span class="kw">=</span> <span class="no">gene_name</span>,<span class="kw">view_matrix</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

<span class="no">csp2</span></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-9-1.png" width="100%"></p>
<div id="display-corr-value-1" class="section level3">
<h3 class="hasAnchor">
<a href="#display-corr-value-1" class="anchor"></a>Display corr value</h3>
<div class="sourceCode" id="cb7"><pre class="r"><span class="no">cor_tbl3</span> <span class="kw">&lt;-</span> <span class="no">cor_tbl2</span> <span class="kw">%&gt;%</span> <span class="kw pkg">dplyr</span><span class="kw ns">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">`Rep.A`</span> <span class="kw">==</span> <span class="no">`Rep.B`</span>)
<span class="no">csp3</span> <span class="kw">&lt;-</span> <span class="kw pkg">corplot</span><span class="kw ns">::</span><span class="fu"><a href="reference/get_pair_wise_scatter.html">get_pair_wise_scatter</a></span>(<span class="kw">dat_tbl</span> <span class="kw">=</span> <span class="no">expr_mat</span>, <span class="kw">group_tbl</span> <span class="kw">=</span> <span class="no">groups</span>,<span class="kw">var_plot</span> <span class="kw">=</span> <span class="no">condition</span>, <span class="kw">var_plot_group</span> <span class="kw">=</span> <span class="no">repl</span>,<span class="kw">dat_id</span> <span class="kw">=</span> <span class="no">gene_name</span>,<span class="kw">view_matrix</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

<span class="no">csp2</span> + <span class="kw pkg">ggplot2</span><span class="kw ns">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span>(<span class="kw">data</span> <span class="kw">=</span> <span class="no">cor_tbl3</span>,  <span class="kw">x</span> <span class="kw">=</span> <span class="fl">3</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="fl">18</span>, <span class="kw pkg">ggplot2</span><span class="kw ns">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">label</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"r"</span>,<span class="st">"="</span>,<span class="no">corr</span> , <span class="kw">sep</span> <span class="kw">=</span> <span class="st">""</span>)) ,
                         <span class="kw">fontface</span><span class="kw">=</span><span class="st">"italic"</span> , <span class="kw">col</span> <span class="kw">=</span> <span class="st">"red"</span>)</pre></div>
<p><img src="reference/figures/README-unnamed-chunk-10-1.png" width="100%"></p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/cparsania/corplot/">https://​github.com/​cparsania/​corplot/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/cparsania/corplot/issues">https://​github.com/​cparsania/​corplot/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Chirag Parsania <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/cparsania/corplot"><img src="https://img.shields.io/badge/devel%20version-0.1.0-blue.svg"></a></li>
<li><a href="https://www.tidyverse.org/lifecycle/#experimental"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Chirag Parsania.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
